### PR TITLE
Basic implementation of ddc-stub-service

### PR DESCRIPTION
**Pull request by kiranpl**

### CODE REVIEW

File: app/uk/gov/hmrc/ddc_stub_service/controllers/ReferenceLookupController.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import uk.gov.hmrc.ddc_stub_service.utils._
+import play.api.mvc._
+import play.api.Environment
+import javax.inject.{Inject, Singleton}
+import scala.io.Source
+
+@Singleton()
+class ReferenceLookupController @Inject()(environment: Environment, cc: ControllerComponents)
+  extends BackendController(cc) {
+
```
Comment (roccobruno) [inline app/uk/gov/hmrc/ddc_stub_service/controllers/ReferenceLookupController.scala:29]: Should there be also something to know whether is a call for the Pega UI or the Pega Customer  Output?

File: conf/resources/data/data/5330.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "5330":  [
+    {"7006":  "Child Benefit Debt"},
+    {"7010":  "Guardians Allowance GB Debt"},
```
Comment (roccobruno) [inline conf/resources/data/data/5330.json:4]: I think we also need a description for 'mainTrans'

**Review by roccobruno — COMMENTED** [review]
This is good. Any plan to make it dynamic??? I think from a testing point of view we ll need it, or not?

File: app/uk/gov/hmrc/ddc_stub_service/controllers/CaseManagementSystemController.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  private val basePath = "conf/resources/data"
+  private val casePath = "/debt/"
+
+  private val errorUtr = "5005005005"
+
+  private val listHelper: ListHelper = new ListHelper()
+
+  def getCaseDetails(UTR: String, duties: Option[String]) = Action {
+    checkForError(
```
Comment (roccobruno) [inline app/uk/gov/hmrc/ddc_stub_service/controllers/CaseManagementSystemController.scala:38]: maybe using UTR is confusing?? we are not gonna pass that.. but just an identifier for the 'debt'.. something like `uniqueItemReference` would do

File: conf/resources/data/debt/debt002.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "duties": [
+      {
+        "debtItemChargeId": "duty01",
+        "originalAmount": 300000,
+        "mainTrans": "1085",
+        "subTrans": "1000",
+        "interestStartDate": "2018-06-10",
+        "dateCreated": "2018-06-10",
+        "dateCalculationTo": "2020-02-10",
+        "paymentHistory": []
```
Comment (roccobruno) [inline conf/resources/data/debt/debt002.json:12]: u also need to return `breathingSpace` bits.. Otherwise IFS will complain

**Review by roccobruno — COMMENTED** [review]
Also don't forget the service manager bit.. :)

**Review by beckfordp — APPROVED** [review]
LGTM 👍

**Review by roccobruno — APPROVED** [review]

File: app/uk/gov/hmrc/ddc_stub_service/controllers/MicroserviceHelloWorldController.scala [hunk 5]
```diff
@@ -1,3 +1,19 @@
+/*
```
[the remark this answers is not in the document]
Comment (kiranpl) [reply app/uk/gov/hmrc/ddc_stub_service/controllers/MicroserviceHelloWorldController.scala:1]: /hello-world endpoint helps to check if the service is up and running

File: conf/app.routes [hunk 6]
```diff
@@ -1,3 +1,7 @@
 # microservice specific routes
 
 GET        /hello-world             uk.gov.hmrc.ddc_stub_service.controllers.MicroserviceHelloWorldController.hello
+GET        /debt                    uk.gov.hmrc.ddc_stub_service.controllers.CaseManagementSystemController.getList
+GET        /debt/:uniqueReference   uk.gov.hmrc.ddc_stub_service.controllers.CaseManagementSystemController.getCaseDetails(uniqueReference: String, duties: Option[String])
```
[the remark this answers is not in the document]
Comment (kiranpl) [reply conf/app.routes]: @TevynAllen , yes I agree, this has not been implemented and is WIP

**Pull request merged**